### PR TITLE
Adjust to let total soil thickness and max root depth be varied and not fixed

### PR DESCRIPTION
**Pull request by MostafaGomaa93**
### Description
The indices of the first 4 layers (layers close to land surface) are from 51 to 55 (as long as the total soil thickness equals 5 m). However, if we change the total soil thickness (not 5 m), then the indices of the first 4 layers will not be from 51 to 55 anymore. So, a simple change in these 3 files ([one](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/f96ca1ca8176cab732b5ba440ec206a988aeca31/src/calc_sfactor.m#L2C5-L2C58), [two](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/f96ca1ca8176cab732b5ba440ec206a988aeca31/src/calc_rsoil.m#L5C4-L5C48), [three](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/f96ca1ca8176cab732b5ba440ec206a988aeca31/src/%2Bsoilmoisture/calculateBoundaryConditions.m#L75)) to get the proper indices of the first 4 layers. These changes are related to issue [237](https://github.com/EcoExtreML/STEMMUS_SCOPE/issues/237). 

### Checklist
- [x] Add a reference to related issues.
- [x] @mentions of the person or team responsible for reviewing proposed changes.
- [x] This pull request has a descriptive title.
- [x] Code is written according to the [guidelines](http://cnl.sogang.ac.kr/cnlab/lectures/programming/matlab/Richard_Johnson-MatlabStyle2_book.pdf).
- [x] The checks by [MISS_HIT style checker and linter](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/CONTRIBUTING.md#follow-matlab-style-guidelines), below the pull request, are successful (green).
- [x] Documentation is available.
- [x] Add changes to the [changelog file](CHANGELOG.md) under section `Unreleased`.
- [x] Model runs successfully.
- [ ] Ask a meinatainer to re-generate exe file if matlab codes are changed. About how to create an exe file, see [exe readme](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/run_model_on_snellius/exe/README.md).

### CODE REVIEW

File: src/+soilmoisture/calculateBoundaryConditions.m [hunk 1]
```diff
@@ -72,7 +72,7 @@
         % Check applied infiltration doesn't exceed infiltration capacity
         topThick = 5; % first 5 cm of the soil
         satCap = SoilProperties.theta_s0 * topThick; % saturation capacity represented by saturated water content of the top 5 cm of the soil
-        actTheta = ModelSettings.DeltZ(51:54) * SoilVariables.Theta_UU(51:54, 1); % actual moisture of the top 5 cm of the soil
+        actTheta = ModelSettings.DeltZ(end - 4:end - 1) * SoilVariables.Theta_UU(end - 4:end - 1, 1); % actual moisture of the top 5 cm of the soil
```
Comment (SarahAlidoost) [inline src/+soilmoisture/calculateBoundaryConditions.m:75]: Here you are assuming that the lengths of all changed variables are 55. But the length of `ModelSettings.DeltZ` is 54 similar to `ModelSettings.NL`. So,` end - 4:end - 1` is not the same as `51:54` for `ModelSettings.DeltZ`.

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@MostafaGomaa93 please see my comment on the size of ModelSettings.DeltZ. Also, can you add one line to the changelog file explaining the changes? Thanks. After fixing the code, I will re-generate the exe file.

File: src/+soilmoisture/calculateBoundaryConditions.m (hunk 1 above)
Comment (MostafaGomaa93) [reply src/+soilmoisture/calculateBoundaryConditions.m:75]: Thanks for catching this mistake. you are right, `ModelSettings.DeltZ` length is 54 while `SoilVariables.Theta_UU` is 55. so the line should be corrected to

`actTheta = ModelSettings.DeltZ(end - 3:end) * SoilVariables.Theta_UU(end - 4:end - 1, 1);`

**Comment by MostafaGomaa93** [comment]
> @MostafaGomaa93 please see my comment on the size of ModelSettings.DeltZ. Also, can you add one line to the changelog file explaining the changes? Thanks. After fixing the code, I will re-generate the exe file.

Could you show me how can I make changes in the changelog file and where is that file?

**Comment by SarahAlidoost** [comment]
> > @MostafaGomaa93 please see my comment on the size of ModelSettings.DeltZ. Also, can you add one line to the changelog file explaining the changes? Thanks. After fixing the code, I will re-generate the exe file.
> 
> Could you show me how can I make changes in the changelog file and where is that file?

The [CHANGELOG.md](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/SSM_v.0.3.3/CHANGELOG.md ) is in the root of `STEMMUS_SCOPE` folder. In the file, under the section **Unreleased/fixed**, you can add one item for explaining changes in this pull request. Here example:

```markdown
# Unreleased

**Changed:**

- ....

**Fixed:**

- .....
- The issue of defining te indices of first four layers.... is fixed in .... 
```

**Comment by MostafaGomaa93** [comment]
> > > @MostafaGomaa93 please see my comment on the size of ModelSettings.DeltZ. Also, can you add one line to the changelog file explaining the changes? Thanks. After fixing the code, I will re-generate the exe file.
> > 
> > 
> > Could you show me how can I make changes in the changelog file and where is that file?
> 
> The [CHANGELOG.md](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/SSM_v.0.3.3/CHANGELOG.md) is in the root of `STEMMUS_SCOPE` folder. In the file, under the section **Unreleased/fixed**, you can add one item for explaining changes in this pull request. Here example:
> 
> ```
> # Unreleased
> 
> **Changed:**
> 
> - ....
> 
> **Fixed:**
> 
> - .....
> - The issue of defining te indices of first four layers.... is fixed in .... 
> ```

Thanks, I did it already

**Review by SarahAlidoost — APPROVED** [review]
@MostafaGomaa93 thanks for the fixes :+1:

**Pull request merged**